statsmodels/tsa/statespace/mlemodel.py

```diff
@@ -0,0 +1,797 @@
+    def __init__(self, model, *args, **kwargs):
+        self.data = model.data
+
+        # Save the model output
+        self._endog_names = model.endog_names
+        self._exog_names = model.endog_names
+        self._params = model.params
+        self._params_names = model.params_names
+        self._model_names = model.model_names
+        self._model_latex_names = model.model_latex_names
+
+        # Associate the names with the true parameters
+        params = pd.Series(self._params, index=self._params_names)
+
+        # Initialize the Statsmodels model base
+        tsbase.TimeSeriesModelResults.__init__(self, model, params,
+                                               normalized_cov_params=None,
+                                               scale=1., *args, **kwargs)
+
+        # Initialize the statespace representation
+        super(MLEResults, self).__init__(model)
+
+        # Setup the cache
+        self._cache = resettable_cache()
+
+    @cache_readonly
+    def aic(self):
+        # return -2*self.llf + 2*self.params.shape[0]
+        return aic(self.llf, self.nobs, self.params.shape[0])
+
+    @cache_readonly
+    def bic(self):
+        # return -2*self.llf + self.params.shape[0]*np.log(self.nobs)
+        return bic(self.llf, self.nobs, self.params.shape[0])
+
+    @cache_readonly
+    def bse(self):
+        return np.sqrt(np.diagonal(self.cov_params))
+
+    @cache_readonly
+    def cov_params(self):
+        hessian = self.model.hessian(
+            self._params, set_params=False, transformed=True,
+            initial_state=self.initial_state,
+            initial_state_cov=self.initial_state_cov
+        )
+
+        # Reset the matrices to the saved parameters (since they were
+        # overwritten in the hessian call)
+        self.model.update(self.model.params)
+
+        return -np.linalg.inv(hessian*self.nobs)
+
+    @cache_readonly
+    def cov_params_delta(self):
+        # Uses Delta method (method of propagation of errors)
+
+        unconstrained = self.model.untransform_params(self._params)
+        jacobian = self.model.transform_jacobian(unconstrained)
+        hessian = self.model.hessian(
+            unconstrained, set_params=False,
+            initial_state=self.initial_state,
+            initial_state_cov=self.initial_state_cov
+        )
+
+        # Reset the matrices to the saved parameters (since they were
+        # overwritten in the hessian call)
+        self.model.update(self.model.params)
+
+        return jacobian.dot(-np.linalg.inv(hessian*self.nobs)).dot(jacobian.T)
+
+    def fittedvalues(self):
+        """The predicted values of the model."""
+        return self.forecasts.copy()
+
+    @cache_readonly
+    def hqic(self):
+        # return -2*self.llf + 2*np.log(np.log(self.nobs))*self.params.shape[0]
+        return hqic(self.llf, self.nobs, self.params.shape[0])
+
+    @cache_readonly
+    def llf(self):
+        return self.loglikelihood[self.loglikelihood_burn:].sum()
```

Comment:
I think I misunderstood your point, since it looks like all results classes use `llf()`. `self.loglikelihood` is an array; I think it's essentially what would be returned by `loglikeobs()`.